odoo/addons/base/models/ir_actions_report.py

```diff
@@ -524,12 +525,22 @@ def _run_wkhtmltopdf(
 
         files_command_args = []
         temporary_files = []
+        temp_session = None
 
         # Passing the cookie to wkhtmltopdf in order to resolve internal links.
-        if request and request.db:
+        if request and request.db and request.session.sid:
+            # Create a temporary session which will not create device logs
+            temp_session = root.session_store.new()
+            temp_session.update({
+                **request.session,
+                '_trace_disable': True,
+            })
+            temp_session.session_token = security.compute_session_token(temp_session, self.env)
```

Comment:
Shouldn't the session token only be set when `session.uid != None` (public user)

Reply:
In fact, we shouldn't compute a session token when we don't have an `uid` in the session !
Thanks for the remark